common/lib/xmodule/xmodule/js/src/word_cloud/word_cloud_main.js

```diff
@@ -154,136 +154,176 @@
+                    this.width / Math.abs(bounds[0].x - this.width / 2),
+                    this.height / Math.abs(bounds[1].y - this.height / 2),
+                    this.height / Math.abs(bounds[0].y - this.height / 2)
+                );
             }
 
             $.each(response.student_words, function(word, stat) {
                 var percent = (response.display_student_percents) ? ' ' + (Math.round(100 * (stat / response.total_count))) + '%' : '';
 
-                studentWordsKeys.push('<strong>' + word + '</strong>' + percent);
+                studentWordsKeys.push(HtmlUtils.interpolateHtml(
+                    '{listStart}{startTag}{word}{endTag}{percent}{listEnd}',
+                    {
+                        listStart: HtmlUtils.HTML('<li>'),
+                        startTag: HtmlUtils.HTML('<strong>'),
+                        word: word,
+                        endTag: HtmlUtils.HTML('</strong>'),
+                        percent: percent,
+                        listEnd: HtmlUtils.HTML('</li>')
+                    }
+                ).toString());
             });
-            studentWordsStr = '' + studentWordsKeys.join(', ');
+
+            studentWordsStr = '' + studentWordsKeys.join('');
 
             cloudSectionEl
-            .addClass('active')
-            .find('.your_words').html(studentWordsStr)
-            .end()
-            .find('.total_num_words').html(response.total_count);
+                .addClass('active');
+
+            HtmlUtils.setHtml(
+                cloudSectionEl.find('.your_words'),
+                HtmlUtils.HTML(studentWordsStr)
+            );
+
+            cloudSectionEl.find('.your_words').end().find('.total_num_words')
+                .text(response.total_count);
 
             $(cloudSectionEl.attr('id') + ' .word_cloud').empty();
 
-        // Actual drawing of word cloud.
+            // Actual drawing of word cloud.
             groupEl = d3.select('#' + cloudSectionEl.attr('id') + ' .word_cloud').append('svg')
-                .attr('width', this.width)
-                .attr('height', this.height)
-                .append('g')
-                .attr('transform', 'translate(' + (0.5 * this.width) + ',' + (0.5 * this.height) + ')')
-                .selectAll('text')
-                .data(words)
-                .enter().append('g');
+                    .attr('width', this.width)
+                    .attr('height', this.height)
+                    .append('g')
+                    .attr('transform', 'translate(' + (0.5 * this.width) + ',' + (0.5 * this.height) + ')')
+                    .selectAll('text')
+                    .data(words)
+                    .enter()
+                    .append('g')
+                    .attr('aria-describedby', function(d) {
+                        return HtmlUtils.interpolateHtml(
+                            gettext('text_word_{word} title_word_{word}'),
```

Comment:
Took me a few hours, but I figured this out. I'm using a unique ID per `g` that is a combination of the word cloud's `id` and a randomly generated number between 1 and 999. This works though the `id`'s are long.